src/kernel/imfilter_opencl.jl

```diff
@@ -0,0 +1,94 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+const array_kernel(::OpenCLMethod, array) = array
+
+const view_kernel(::OpenCLMethod, array, I) = view(array, I)
+
+function imfilter_kernel(::OpenCLMethod, img, krn)
+  # retrieve basic info
+  T = ComplexF64
+
+  # retrieve OpenCL info
+  device, ctx, queue = cl.create_compute_context()
+
+  # build OpenCL program kernels
+  conj_kernel = build_conj_kernel(ctx)
+  mult_kernel = build_mult_kernel(ctx)
+
+  # pad img to support CLFFT operations
+  padimg = pad_opencl_img(img)
+
+  # pad krn to common size with padimg
+  padkrn = zeros(eltype(img), size(padimg))
```

Comment:
There is no way to allocate this directly on the GPU as we did with the CUDA? Maybe something along the lines of the `bufres` initialization below?

Reply:
I think we can. But I need to check how OpenCL.j works with a matrix, it doesn't implements a copyto! function

Reply:
No worries then, I think we can proceed with the current code. Can this branch already be used for our case study? Should we move to the platform-aware version?

Reply:
Yes, I think so